sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -220,22 +220,30 @@ public static OffsetDateTime parseDate(String dateString) {
      * @param data The input data which needs to convert to ByteBuffer.
      * @param length The expected input data length.
      * @param blockSize The size of each ByteBuffer.
+     * @param markAndReset Whether the stream needs to be marked and reset. This should generally always be true to
+     * support retries. It is false in the case of buffered upload to support non markable streams because buffered
+     * upload uses its own mechanisms to support retries.
      * @return {@link ByteBuffer} which contains the input data.
      * @throws UnexpectedLengthException when input data length mismatch input length.
      * @throws RuntimeException When I/O error occurs.
      */
-    public static Flux<ByteBuffer> convertStreamToByteBuffer(InputStream data, long length, int blockSize) {
-        data.mark(Integer.MAX_VALUE);
+    public static Flux<ByteBuffer> convertStreamToByteBuffer(InputStream data, long length, int blockSize,
```

Comment:
Given this is a public API should make this a new overload and have the previous method pass the default case which appears to be `true`.